### ISSUE TITLE
Migrate existing datastores to use `combineStores` utility

### ISSUE THREAD

**Issue by felixarntz** [google/site-kit-wp#1502]
After #1400, all existing stores should rely on the `combineStores` utility instead of the individual `collect...` utilities.

**The PR for this should be opened against the `1400/enhancement-create-combinestore-utility` branch.**

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All existing datastores should rely on `combineStores` and not use any `collect...` utilities.
* All `collect...` utility functions should no longer be exposed on `Data` / `googlesitekit.data`.

## Implementation Brief

* Replace all usages of the `collect...` utilities with `combineStores`.
* Remove the `collect...` utilities from being exposed on the `Data` object in `assets/js/googlesitekit/data/index.js`.

## QA Brief

* 

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
Oops... duplicate of #1476.
